### PR TITLE
Many small improvements

### PR DESCRIPTION
**Pull request by fizista**
Descriptions of the changes can be found in the following descriptions of commits.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -18,3 +18,8 @@
     mode: 0644
     backup: yes
   notify: Restart smartd
+
+- name: Check smartd service is started
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: I don't think that's necessary, there exists a handler that will restart the service if the config changes.

File: templates/smartd.conf.j2 [hunk 2]
```diff
@@ -25,11 +25,15 @@
 	{%- if config.test_schedule | d(smartd_default_test_schedule) %}
 		{{- " -s " }}{{ config.test_schedule | d(smartd_default_test_schedule) -}}
 	{%- endif %}
-	{%- if 'mail_recipients' in config %}
-		{{- " -m " }}{{ config.mail_recipients | join(",") }}
+	{%- if 'mail_recipients' in config or smartd_default_mail_recipients %}
+        {%- if 'mail_recipients' in config %}
```
Comment (SlothOfAnarchy) [inline templates/smartd.conf.j2]: You can remove the inner if clause and use the default filter like this:
```
{{- " -m " }}{{ config.mail_recipients | default(smartd_default_mail_recipients) | join(",") -}}
```

File: defaults/main.yml [hunk 3]
```diff
@@ -16,6 +16,8 @@ smartd_default_non_smart_settings:
   - lookahead,on
 smartd_default_test_schedule: (L/../../6/01|S/../.././02)
 smartd_default_mail_frequency: diminishing
+smartd_default_mail_script:
```
Comment (SlothOfAnarchy) [inline defaults/main.yml:19]: Shouldn't these be an empty list?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 1 above)
Comment (fizista) [reply tasks/main.yml]: Yes, you are right. I've added configuration validation instead of this code.

File: templates/smartd.conf.j2 (hunk 2 above)
Comment (fizista) [reply templates/smartd.conf.j2]: I corrected it and checked if it works. It is OK.

File: defaults/main.yml (hunk 3 above)
Comment (fizista) [reply defaults/main.yml:19]: Yes, it doesn't disturb anything. However, there is information in the file about the variable that can be assigned.

The code works well with an empty variable. I checked it.

File: tasks/main.yml (hunk 1 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml]: Cool, I didn't know of this until now!

File: defaults/main.yml (hunk 3 above)
Comment (SlothOfAnarchy) [reply defaults/main.yml]: Can you change it to an empty list so the reader knows of which type the variable should be?

File: defaults/main.yml [hunk 4]
```diff
@@ -16,6 +16,8 @@ smartd_default_non_smart_settings:
...
+smartd_default_mail_recipients:
```
Comment (SlothOfAnarchy) [inline defaults/main.yml]: Can you change this to an empty list so the reader knows of which type the variable should be?

```suggestion
smartd_default_mail_recipients: []
```

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 4 above)
Comment (fizista) [reply defaults/main.yml]: Corrected.

**Review by SlothOfAnarchy — APPROVED** [review]

File: defaults/main.yml (hunk 4 above)
Comment (SlothOfAnarchy) [reply defaults/main.yml]: Merged. For future reference: Suggested changes can be accepted in the GitHub UI - that way you don't have to commit the same change manually.

**Comment by SlothOfAnarchy** [comment]
Thank you for your contribution! You may find interest in some other roles of our organization. The ones we use in production are well-maintained.

**Pull request merged**